src/plugins/data/common/query/to_expression_ast.ts

```diff
@@ -52,16 +54,25 @@ export async function queryStateToExpressionAst({
     filters: filters && filtersToAst(filters),
   });
   const ast = buildExpression([kibana, kibanaContext]).toAst();
-
   if (query && isOfAggregateQueryType(query)) {
     const mode = getAggregateQueryMode(query);
     // sql query
     if (mode === 'sql' && 'sql' in query) {
       const idxPattern = getIndexPatternFromSQLQuery(query.sql);
       const idsTitles = await dataViewsService.getIdsWithTitle();
-      const dataViewIdTitle = idsTitles.find(({ title }) => title === idxPattern);
-      if (dataViewIdTitle) {
-        const dataView = await dataViewsService.get(dataViewIdTitle.id);
+
+      let dataViewId = idsTitles.find(({ title }) => title === idxPattern)?.id;
```

Comment:
ad hoc dataview should be in cache at this point, we shouldn't be passing in a separate list of dataviews to use

Reply:
So in text based query languages we have to extract the actual index pattern of the given sql text, so to allow the users to make use of the current ad hoc data views, we need to have a list of those ..tbc

Reply:
in sql mode there is no dataview selection. why would we want to tie this with ad hoc dataviews ? 
imo, adhoc dataviews should not be supported in sql mode.


actually, imo we shouldn't do this matching of the string user entered to the dataview (not even for persisted ones).
its very likely that user is going to get unwanted results. For example we have a dataview `kib*`, user enters `select * from kibana`, `select * from k*`, .... it all works, without the dataview present. now he enters `select * from kib*` and suddently dataview kicks in and his results change without him understanding why.

Reply:
@ppisljar i see, well, if I have a ad hoc one selected, switch to the SQL mode, as a user I'd expect this to work. since SQL is technical preview so we can change the behavior in this area, you're fine moving forward with this PR?

Reply:
yes, sql is in preview but this does change our public apis. i am still reviewing the pr, but i am fine with this being ignored for now. However I could see how the fact that sql is in preview could be a good argument to not implement this for now rather than implement it and change it later.

Reply:
I'm fine with resolving the whole ad-hoc data view <-> textbased querying interaction later on, there are big unresolved questions anyway.

Reply:
if condition should still be added (if no dataview is found, do not try to access dataView.timefield)

Reply:
I agree, let's not implement the ad-hoc --> text based languages transition for now. We can:
- either allow it but an error will occur (Matthias video)
- disable the SQL menu if adhoc dataview is selected

Although the second one doesn't require a lot of development I am also fine to keep the behavior as it is on the video (SQL is on tech preview after all) and implement the second one on another PR.

Wdyt @flash1293 and @ppisljar ?

Reply:
Yes the reason that we have the `throw new Error(`No data view found for index pattern ${idxPattern}`);` is because we want the SQL mode to work only if dataview is detected. We could move it on the application (discover, lens) but this means that each app has to add logic for it.

Reply:
ok we all fine @ppisljar @stratoula @flash1293 with reverting this change, so it's an error message when switching from an ad-hoc data view to SQL?

Reply:
Updated.